### PR TITLE
Fixing missing language code

### PR DESCRIPTION
**Pull request by artur-stepien**
Pull Request for Issue #28963 .

### Summary of Changes
Just fixing the #28963

### Testing Instructions
Look #28963

### Expected result
Look #28963

### Actual result
Look #28963

### Documentation Changes Required
No

### CODE REVIEW

**Comment by artur-stepien** [comment]
Hold on with the PR as I might found an issue with this.

File: layouts/joomla/html/batch/item.php [hunk 1]
```diff
@@ -28,7 +28,11 @@
 <div id="batch-choose-action" class="control-group">
 	<select name="batch[category_id]" class="inputbox" id="batch-category-id">
 		<option value=""><?php echo JText::_('JLIB_HTML_BATCH_NO_CATEGORY'); ?></option>
-		<?php echo JHtml::_('select.options', JHtml::_('category.options', $extension)); ?>
+		<?php if( isset($addRoot) and $addRoot ): ?>
```
Comment (richard67) [inline layouts/joomla/html/batch/item.php]:
```suggestion
		<?php if (isset($addRoot) && $addRoot) : ?>
```

**Comment by artur-stepien** [comment]
Ok, it should be fine now. There was a missing root option in com_categories when used in com_categories itself. Now there is just a layout option for that thing if any of the extensions need a root option in the category batch pop-up. Also, in the batch pop-up I change JHtml to JLayout instead cause modifying JHtml function felt wrong.

**Comment by infograf768** [comment]
This does not work here. I don't get the language tag between parenthesis after the cat name.
Please add a screenshot of what you get (or want to get) after this PR.

**Comment by artur-stepien** [comment]
@infograf768 Give me few hours. While adding missing "Add to root" option I removed the fix I made in the first place :D Need to find a solution to reconcile both of the fixes

**Comment by infograf768** [comment]
You are failing because the change is very simple and only concerns one file:
```
diff --git a/libraries/cms/html/category.php b/libraries/cms/html/category.php
index 413c246..0d7b76d 100644
--- a/libraries/cms/html/category.php
+++ b/libraries/cms/html/category.php
@@ -155,5 +155,5 @@
 			$db = JFactory::getDbo();
 			$query = $db->getQuery(true)
-				->select('a.id, a.title, a.level, a.parent_id')
+				->select('a.id, a.title, a.level, a.parent_id, a.language')
 				->from('#__categories AS a')
 				->where('a.parent_id > 0');
@@ -192,4 +192,10 @@
 				$repeat = ($item->level - 1 >= 0) ? $item->level - 1 : 0;
 				$item->title = str_repeat('- ', $repeat) . $item->title;
+
+				if ($item->language !== '*')
+				{
+					$item->title .= ' (' . $item->language . ')';
+				}
+
 				static::$items[$hash][] = JHtml::_('select.option', $item->id, $item->title);
 			}
```

In the screen shot below the only change is adding (xx-XX) after the name as I already added in some of the categories names the lang tag.

Screen Shot 2020-05-08 at 09 07 31

I can make new PR and delete this one.

**Comment by artur-stepien** [comment]
Yeah, that was the missing part to keep both changes. But now I'm interested if there is a point to keep both category.options and category.categories cause they both seems to do the same. but with or without root item.

**Comment by infograf768** [comment]
I would not merge this PR.
The change I proposed is enough. Not necessary to modify other files.
If you want to refactor totally this feature, please propose it for 4.0.

**Comment by artur-stepien** [comment]
@infograf768 So do it your way.

**Comment by infograf768** [comment]
I created the PR https://github.com/joomla/joomla-cms/pull/29003

I suggest you make a specific PR for the rest vs J4

**Comment by infograf768** [comment]
@artur-stepien 
I will make specific PR for J4 using your code adapted to J4 as layout helpers are used for all batch and there are no specific differenciation between category.categories and category.options

**Comment by artur-stepien** [comment]
@infograf768 Thanks.

**Comment by infograf768** [comment]
Can be tested here
https://github.com/joomla/joomla-cms/pull/29053

**Pull request closed without merging**